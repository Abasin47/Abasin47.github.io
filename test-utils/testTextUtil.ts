export const testInitalMessages = {
    'test.text': 'test text',
    'test.tooltip': 'test tooltip',
    'test.dummy': 'test dummy',
    'test.key': 'test key'
};
