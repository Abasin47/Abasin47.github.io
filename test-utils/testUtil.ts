/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import { render, queries, RenderOptions } from '@testing-library/react';
import { ReactElement } from 'react';

const customRender = (
    ui: ReactElement,
    options?: Omit<RenderOptions, 'queries'>
) => render(ui, { queries: { ...queries }, ...options });

export * from '@testing-library/react';
export { customRender as render };
