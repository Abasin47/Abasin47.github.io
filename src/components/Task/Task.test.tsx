import * as React from 'react';

import { render } from '../../../test-utils/testUtil';
import { Task } from './Task';

import testImg from '../../assets/images/deployment.svg';

describe(Task, () => {
    test('snapshot', () => {
        const wrapper = render(
            <Task
                img={testImg}
                imgPlaceholder={testImg}
                intlPrefix="test.text"
                completed={true}
            />
        );
        expect(wrapper.asFragment()).toMatchSnapshot();
    });
});
