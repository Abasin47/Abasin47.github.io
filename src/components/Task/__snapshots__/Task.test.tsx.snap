// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task snapshot 1`] = `
<DocumentFragment>
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  padding: 1rem 20%;
  box-shadow: 0px -1px 0px 0px #4a4a4a;
  background-color: #CEF0C4;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
  margin-top: 1rem;
}

.c3 {
  height: 8rem;
  margin: 0 0 0 1rem;
}

.c1 {
  font-size: 1.5rem;
  text-align: end;
  color: #4a4a4a;
}

.c4 {
  color: #4a4a4a;
  margin: auto 0;
}

<section
    class="c0"
  >
    <h3
      class="c1"
    >
      test.text.title
    </h3>
    <div
      class="c2"
    >
      <img
        class="c3"
      />
      <p
        class="c4"
      >
        test.text.text
      </p>
    </div>
  </section>
</DocumentFragment>
`;
